src/maps/hashmap.js

```diff
@@ -0,0 +1,134 @@
+const IAMap = require('iamap')
+const murmurhash3 = require('murmurhash3js-revisited')
+const Type = require('../base')
+const Block = require('@ipld/block')
+
+const _type = 'IPLD/Experimental/HashMap/0'
+
+class HashMap extends Type {
+  get _type () {
+    return _type
+  }
+
+  get kind () {
+    return 'map'
+  }
+
+  async get (args, continuation) {
+    let path = args.path.split('/').filter(Boolean)
+    let key = path.shift()
+
+    if (!key) {
+      throw new Error('HashMap type needs a string to look up')
+    }
+
+    // TODO: this should be a validation of this.node against the HAMT root block schema
+    if (!IAMap.isRootSerializable(this.node) || !IAMap.isSerializable(this.node)) {
+      throw new Error('Block does not contain expected HashMap root data')
+    }
+
+    // provide a store to IAMap that won't do any retrieval but will intercept load() calls
+    // so we can pass that up for continuation
```

Comment:
IAMap assumes it has a backing store with `save()`, `load()` and an `isEqual(id1, id2)` as it's agnostic to storage system and what an "id" is. Mutation and fetch operations work on the storage system as it needs to save new blocks or load existing ones. So in the get() situation we're only going to load blocks and loading here means passing CIDs and continuations, hence this `fauxStorage` thing which does the intercepting.

Reply:
Ah, clever. Once we have the write side of this figured out we should be able to migrate the whole implementation over and no longer need this but this is a nice hack in the interim.